client/js/components/SelfGuidedItem/SelfGuidedItem.js

```diff
@@ -24,54 +24,64 @@ const SelfGuidedItem = ({selfGuidedItem, navigation}) => (
+                  </View>
+                  <View style={styles.flexColumn}>
+                    <Text style={styles.subTitles}>Distance</Text>
+                    <Text style={styles.infobox}>
+                      {selfGuidedItem.distance}
+                    </Text>
+                  </View>
+                  <View style={styles.flexColumn}>
+                    <Text style={styles.subTitles}>Duration</Text>
+                    <Text style={styles.infobox}>
+                      {selfGuidedItem.duration} hours
+                    </Text>
+                  </View>
                 </View>
-                <View style={styles.flexColumn}>
-                  <Text style={styles.subTitles}>Distance</Text>
-                  <Text style={styles.infobox}>{selfGuidedItem.distance}</Text>
-                </View>
-                <View style={styles.flexColumn}>
-                  <Text style={styles.subTitles}>Duration</Text>
-                  <Text style={styles.infobox}>
-                    {selfGuidedItem.duration} hours
-                  </Text>
-                </View>
-              </View>
-              <View style={styles.flexRow}>
-                <Image
-                  style={styles.contactSocialMediaIcon}
-                  source={require('../../assets/images/icFaveShareDefault.png')}
-                />
-                <TouchableOpacity
-                  onPress={async e => {
-                    try {
-                      if (
-                        savedIds.some(savedId => savedId === selfGuidedItem.id)
-                      ) {
-                        removeSaveId(selfGuidedItem.id);
-                      } else {
-                        addSaveId(selfGuidedItem.id);
-                      }
-                    } catch (error) {}
-                  }}>
+                <View style={styles.flexRow}>
                   <Image
-                    style={styles.contactSocialMediaIcon}
-                    source={
-                      savedIds.some(savedId => savedId === selfGuidedItem.id)
-                        ? require('../../assets/images/icFaveCircleActive.png')
-                        : require('../../assets/images/icFaveCircleDefault.png')
-                    }
+                    style={styles.contactSocialMediaIconShare}
+                    source={require('../../assets/images/icFaveShareDefault.png')}
                   />
-                </TouchableOpacity>
+                  <TouchableOpacity
+                    onPress={async e => {
+                      try {
+                        if (
+                          savedIds.some(
+                            savedId => savedId === selfGuidedItem.id,
+                          )
+                        ) {
+                          removeSaveId(selfGuidedItem.id);
+                        } else {
+                          addSaveId(selfGuidedItem.id);
+                        }
+                      } catch (error) {}
```

Comment:
What should happen when there is an error? 
Are there any promises that you need to await? Please clarify.

Reply:
Add a Throw new Error there.